tools/nightly-e2e-tests/nightly_e2e.py

```diff
@@ -0,0 +1,147 @@
+        logs_monitor_template_path='monitors-gen/templates/monitor-logs.tf.src',
+        apm_monitor_template_path='monitors-gen/templates/monitor-apm.tf.src'
+    )
+    _ = renderer.render(monitors=all_monitors)
+
+    Linter.shared.print(strict=False)  # Just print linter events, without aborting
+
+
+def generate_terraform_file(args):
+    """
+    Runs linter (in strict mode) for monitors generation, and writes the generated file.
+    """
+    Linter.shared = Linter()
+
+    all_monitors = read_all_monitors(tests_dir=args.tests_dir)
+
+    renderer = MainTF.load_from_templates(
+        main_template_path='monitors-gen/templates/main.tf.src',
+        logs_monitor_template_path='monitors-gen/templates/monitor-logs.tf.src',
+        apm_monitor_template_path='monitors-gen/templates/monitor-apm.tf.src'
+    )
+    rendered = renderer.render(monitors=all_monitors)
+
+    Linter.shared.print(strict=True)  # Print linter events and abort if any
+
+    with open('monitors-gen/main.tf', 'w') as file:
+        file.write(rendered)  # render to file
+
+    print(f"✅ Generated {len(all_monitors)} monitor(s) in {os.path.abspath('monitors-gen/main.tf')}")
+
+    # Create `monitors-gen/secrets.tf` file if it's missing:
+    if not os.path.isfile('monitors-gen/secrets.tf'):
+        shutil.copyfile('monitors-gen/templates/secrets.tf.src', 'monitors-gen/secrets.tf')
+        print(
+            f"⚠️ Before running Terraform you need to configure secrets in {os.path.abspath('monitors-gen/secrets.tf')}"
+        )
+
+
+if __name__ == "__main__":
+    # Change working directory to `tools/nightly-e2e-tests/`
+    print(f'ℹ️ Launch dir: {sys.argv[0]}')
+    launch_dir = os.path.dirname(sys.argv[0])
+    launch_dir = '.' if launch_dir == '' else launch_dir
+    if launch_dir == './tools/nightly-e2e-tests':
+        print(f'    → changing current directory to: {os.getcwd()}/tools/nightly-e2e-tests')
+        os.chdir('tools/nightly-e2e-tests')
+
+    # Read arguments
+    cli = ArgumentParser()
+    commands = cli.add_subparsers(dest='command')
+
+    lint_command = commands.add_parser('lint')
+    generate_tf_command = commands.add_parser('generate-tf')
+
+    lint_command.add_argument(
+        "--tests-dir",
+        required=True,
+        help="Path to the directory containing E2E test definitions."
+    )
+    generate_tf_command.add_argument(
+        "--tests-dir",
+        required=True,
+        help="Path to the directory containing E2E test definitions."
+    )
+
+    cli_args = cli.parse_args()
+
+    try:
+        # Execute given command
+        Linter.shared = NoOpLinter()
+
+        if cli_args.command == 'lint':
+            lint(cli_args)
+        elif cli_args.command == 'generate-tf':
+            generate_terraform_file(cli_args)
+
+    except Exception as error:
+        print(f'❌ nightly_e2e.py failed: {error}')
+        print('-' * 60)
+        traceback.print_exc(file=sys.stdout)
```

Comment:
I'm not familiar with python practices, but just wondering: should it be `stderr` here instead?

Reply:
It could 👍. It's more Unix than Pythonic discussion on whether we should use `STDOUT` or `STDERR`. I used `stdout` for this tool (similar to few other that we have in `dd-sdk-ios`) as I know that this output is well displayed in Bitrise build log 🙂 and I'm not sure if it will appear there if it's `stderr`. But TBH I never tried as it doesn't seem important.

Reply:
indeed, not that important.